### PR TITLE
Log Format and Tracing Changes for Scholar

### PR DESCRIPTION
**Pull request by bhanutejags**
Fixes #issuenumber ; refs #issuenumber

Create a log format that is easy to parse(using CloudWatch expressions), these logs should also have the necessary context/metadata to categorise or group log lines based on HTTP request ID, on Client IP, on User ID, User Agent etc. This would give us more visibility into errors, for example gathering all lines of logs that originated from the same request or same Client IP address can give a us a greater understanding what went wrong.

Here is an example of method in `ApplicationController` which adds a the Current User's ID to the log as a tag and to the XRay Traces.
``` ruby
around_action :add_tracing_and_logging_metadata

  def add_tracing_and_logging_metadata
    XRay.recorder.current_segment.user = current_user.try(:id) || 0

    logger.tagged("CU: #{current_user.try(:id) || 'UnauthenticatedUser'}") do
      yield
    end
  end
```

Example Log snippets, when an user uses Scholar without logging in:
```
I, [2021-05-14T10:53:34.930424 #9076]  INFO -- : [787097bb-8069-4a86-9c29-325c583ad571] [Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.1 Safari/605.1.15] [rails-scholar] [127.0.0.1] [192.168.33.1] [CU: UnauthenticatedUser]   Rendering hyrax/homepage/index.html.erb within layouts/homepage
I, [2021-05-14T10:53:34.941341 #9076]  INFO -- : [787097bb-8069-4a86-9c29-325c583ad571] [Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.1 Safari/605.1.15] [rails-scholar] [127.0.0.1] [192.168.33.1] [CU: UnauthenticatedUser]   Rendered /home.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/hyrax-2.9.4/app/views/_flash_msg.html.erb (0.5ms)
I, [2021-05-14T10:53:34.947516 #9076]  INFO -- : [787097bb-8069-4a86-9c29-325c583ad571] [Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.1 Safari/605.1.15] [rails-scholar] [127.0.0.1] [192.168.33.1] [CU: UnauthenticatedUser]   Rendered layouts/scholar/_scholar_headline.html.erb (0.8ms)
I, [2021-05-14T10:53:34.961606 #9076]  INFO -- : [787097bb-8069-4a86-9c29-325c583ad571] [Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.1 Safari/605.1.15] [rails-scholar] [127.0.0.1] [192.168.33.1] [CU: UnauthenticatedUser]   Rendered hyrax/homepage/_featured_researcher.html.erb (0.5ms)
I, [2021-05-14T10:53:34.970927 #9076]  INFO -- : [787097bb-8069-4a86-9c29-325c583ad571] [Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.1 Safari/605.1.15] [rails-scholar] [127.0.0.1] [192.168.33.1] [CU: UnauthenticatedUser]   Rendered hyrax/homepage/_featured_collections.html.erb (2.6ms)
I, [2021-05-14T10:53:34.981412 #9076]  INFO -- : [787097bb-8069-4a86-9c29-325c583ad571] [Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.1 Safari/605.1.15] [rails-scholar] [127.0.0.1] [192.168.33.1] [CU: UnauthenticatedUser]   Rendered hyrax/homepage/_featured_works.html.erb (3.0ms)
I, [2021-05-14T10:53:34.981712 #9076]  INFO -- : [787097bb-8069-4a86-9c29-325c583ad571] [Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.1 Safari/605.1.15] [rails-scholar] [127.0.0.1] [192.168.33.1] [CU: UnauthenticatedUser]   Rendered hyrax/homepage/_home_content.html.erb (27.2ms)
```

Changes proposed in this pull request:
* Added a new Log tags, which add some necessary context to each line of log.
  * These include things like a HTTP request ID, Client IP etc.
* Added Current User ID to XRay Traces.
* The default mirror url in the Solr Wrapper seems to be broken, replaced that with an alternate mirror url hosted by Apache Org.

### CODE REVIEW

File: config/environments/production.rb [hunk 1]
```diff
@@ -51,9 +51,6 @@
   # when problems arise.
   config.log_level = :debug
```
Comment (hortongn) [inline config/environments/production.rb:52]: I suggest we set the production log level to :info

**Review by hortongn — CHANGES_REQUESTED** [review]
@bhanutejags Looks good so far!  I made one comment inline about using log level :info in production mode.  Also, be sure to run `bundle exec rubocop` to clean up some style issues.

Is it maybe useful to use something like `if defined? Xray` on the Xray lines so that we can turn off Xray completely by commenting out the aws-xray-sdk gem?

**Comment by hortongn** [comment]
@bhanutejags Here's an example from the log on scholar-qa.uc.edu:
```
I, [2021-06-16T10:31:44.465206 #9343]  INFO -- : [4ef0da96-5c5f-4df1-9b41-c15e9ea5bb58] [1-60ca0b50-49951f742a6133fb7f8a8242] [Mozilla/5.0 (Macintosh; Intel Mac OS X 10_1
5_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.101 Safari/537.36] [scholar-qa] [10.11.157.241] [No-X-Forwarded-For-Header] [CU: 6]   Rendered /srv/apps/scho
lar_capistrano/shared/vendor/bundle/ruby/2.7.0/gems/hyrax-2.9.4/app/views/shared/_locale_picker.html.erb (0.2ms)
```

All looks good I think except every line is reporting `[No-X-Forwarded-For-Header]`.  Do we need to make a tweak to Apache to fix that?

**Comment by bhanutejags** [comment]
> @bhanutejags Here's an example from the log on scholar-qa.uc.edu:
> 
> ```
> I, [2021-06-16T10:31:44.465206 #9343]  INFO -- : [4ef0da96-5c5f-4df1-9b41-c15e9ea5bb58] [1-60ca0b50-49951f742a6133fb7f8a8242] [Mozilla/5.0 (Macintosh; Intel Mac OS X 10_1
> 5_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.101 Safari/537.36] [scholar-qa] [10.11.157.241] [No-X-Forwarded-For-Header] [CU: 6]   Rendered /srv/apps/scho
> lar_capistrano/shared/vendor/bundle/ruby/2.7.0/gems/hyrax-2.9.4/app/views/shared/_locale_picker.html.erb (0.2ms)
> ```
> 
> All looks good I think except every line is reporting `[No-X-Forwarded-For-Header]`. Do we need to make a tweak to Apache to fix that?

Looks like that, but when I tested with a local Apache HTTP Server installation, it did not need any specific configuration beyond that of simple [VirtualHost](https://httpd.apache.org/docs/2.4/vhosts/examples.html).
I'll go over the Scholar QA's Apache HTTP Server Configuration.

**Comment by hortongn** [comment]
Rebase, review, and merge this after xray has been re-added to project

**Comment by scherztc** [comment]
We are going to close this PR, since we are no longer using Amazon tools like X-Ray for logging.

**Pull request closed without merging**